### PR TITLE
Remove/update references to EOL Python versions

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/python/devguide/issues/661.

We've already removed quite a few other old ones, such as instructions for building older Python versions.

----
:books: Documentation preview :books:: https://cpython-devguide--1227.org.readthedocs.build/

### CODE REVIEW

File: development-tools/clinic.rst [hunk 1]
```diff
@@ -834,8 +834,7 @@ To save time, and to minimize how much you need to learn
 to achieve your first port to Argument Clinic, the walkthrough above tells
 you to use "legacy converters".  "Legacy converters" are a convenience,
 designed explicitly to make porting existing code to Argument Clinic
-easier.  And to be clear, their use is acceptable when porting code for
-Python 3.4.
+easier.
```
Comment (ezio-melotti) [inline development-tools/clinic.rst:837]: Maybe this sentence is useful since it's about legacy converters?  @erlend-aasland?

Comment (erlend-aasland) [reply development-tools/clinic.rst:837]: I don't understand this sentence. IIRC, legacy converters are just aliases for "normal" converters. I can take a closer look tomorrow. Ping me again if I take too long.

Comment (hugovk) [reply development-tools/clinic.rst:837]: @erlend-aasland ping :)

Comment (erlend-aasland) [reply development-tools/clinic.rst:837]: I'm fine with removing this. AFAICS, it has no relevance anymore. If it causes great havoc, I'll take the blame :)

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**